discovery(testing): test_synthesize_docs_happy_path fails after #4678 — AnalyzerService calls same llm mock, breaking assert_awaited_once()

## Problem

After #4678 (AnalyzerService) merged, \`test_synthesize_docs_happy_path\` in \`autobot-backend/services/knowledge/test_kb_synthesizer.py:215\` fails:

\`\`\`
AssertionError: Expected chat to have been awaited once. Awaited 2 times.
\`\`\`

**Root cause:** \`KBSynthesizer._run_analyzer()\` passes \`self._llm\` to \`AnalyzerService.analyze_synthesis_run()\`. The test mocks \`llm.chat\` on the shared LLM instance, so it counts both the synthesis call (expected) and the analyzer call (unexpected from test's perspective).

## Reproduction

\`\`\`bash
python3 -m pytest autobot-backend/services/knowledge/test_kb_synthesizer.py::test_synthesize_docs_happy_path -xvs
\`\`\`

## Fix Options

**Option A (preferred):** Patch \`AnalyzerService\` out in \`test_synthesize_docs_happy_path\` since the test is verifying synthesis, not analysis:

\`\`\`python
with patch("services.knowledge.kb_synthesizer.AnalyzerService") as mock_analyzer:
    ...  # existing test body
    llm.chat.assert_awaited_once()  # now only synthesis call
\`\`\`

**Option B:** Change assertion to expect 2 awaits:
\`\`\`python
assert llm.chat.await_count == 2
\`\`\`

Option A is cleaner — the test's scope is synthesis, not analysis.

## Introduced By

PR #4742 / issue #4678

--- a/autobot-backend/services/knowledge/test_kb_synthesizer.py
+++ b/autobot-backend/services/knowledge/test_kb_synthesizer.py
@@ -209,6 +209,9 @@ async def test_synthesize_docs_happy_path(tmp_path):
     llm = _make_llm("Synthesized summary")
     synth = KBSynthesizer(llm_service=llm)
     synth._collection = col
+    # Issue #4785: AnalyzerService (#4678) makes a second llm.chat call; patch it
+    # out so the synthesis-only assertion on assert_awaited_once() stays valid.
+    synth._run_analyzer = AsyncMock()
 
     await synth.synthesize_docs([str(f)])
 

